exec/java-exec/src/test/java/org/apache/drill/exec/physical/unit/TestOutputBatchSize.java

```diff
@@ -2287,6 +2292,341 @@ public void testHashAggMax() throws ExecutionSetupException {
     opTestBuilder.go();
   }
 
+  @Test
+  public void testNestedLoopJoinMultipleOutputBatches() throws Exception {
+    LogicalExpression functionCallExpr = new FunctionCall("equal",
+            ImmutableList.of((LogicalExpression) new FieldReference("c1", ExpressionPosition.UNKNOWN),
+                    (LogicalExpression) new FieldReference("c2", ExpressionPosition.UNKNOWN)),
+            ExpressionPosition.UNKNOWN);
+
+    NestedLoopJoinPOP nestedLoopJoin = new NestedLoopJoinPOP(null, null, JoinRelType.INNER, functionCallExpr);
+    mockOpContext(nestedLoopJoin, initReservation, maxAllocation);
```

Comment:
Just a comment: This test calls `mockOpContext()`, while the other new NLJ tests don't. I tested all with and without this call, and it seems to make no difference anyway ....